lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -68,47 +74,69 @@ import Servant.Client.Core
 import Servant.Links
     ( Link, safeLink )
 
+import qualified Data.Text as T
+
 -- | Creates a new 'NetworkLayer' connecting to an underlying 'Jormungandr'
 -- backend target.
 newNetworkLayer
     :: forall n. ()
     => BaseUrl
     -> IO (NetworkLayer (Jormungandr n) IO)
-newNetworkLayer url = do
+newNetworkLayer  url = do
     mgr <- newManager defaultManagerSettings
-    return $ mkNetworkLayer $ mkJormungandrLayer mgr url
+    return $ mkNetworkLayer (mkJormungandrLayer mgr url)
 
 -- | Wrap a Jormungandr client into a 'NetworkLayer' common interface.
-mkNetworkLayer :: Monad m => JormungandrLayer m -> NetworkLayer t m
+mkNetworkLayer
+    :: Monad m
+    => JormungandrLayer m
+    -> NetworkLayer t m
 mkNetworkLayer j = NetworkLayer
     { networkTip = do
         t <- (getTipId j) `mappingError`
             ErrNetworkTipNetworkUnreachable
         b <- (getBlock j t) `mappingError` \case
-            ErrGetBlockNotFound (BlockId _) ->
+            ErrGetBlockNotFound _ ->
                 ErrNetworkTipNotFound
             ErrGetBlockNetworkUnreachable e ->
                 ErrNetworkTipNetworkUnreachable e
         return $ header b
 
-    , nextBlocks = error "nextBlocks to be implemented"
+    , nextBlocks = \tipHeader ->
+        -- Get the descendants of the tip's /parent/.
+        -- The first descendant is therefore the current tip itself.
+        tail <$> getBlocksAfter (prevBlockHash tipHeader)
+
     , postTx = error "postTx to be implemented"
     }
   where
     mappingError = flip withExceptT
 
+    getBlocksAfter blockHeader = do
+        let count = 10000
+        ids <- getDescendantIds j blockHeader count `mappingError` \case
+            ErrGetDescendantsNetworkUnreachable e ->
+                ErrGetBlockNetworkUnreachable e
+            ErrGetDescendantsParentNotFound e ->
+                ErrGetBlockNotFound . T.pack . show $ e
+        forM ids getBlock'
+      where
+        getBlock' = getBlock j
```

Comment:
Not sure this indirection is any useful :/ ... `forM ids (getBlock j)` is really fine.